staging/src/k8s.io/client-go/tools/cache/thread_safe_store_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"errors"
+	"strconv"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrorFromIndex(t *testing.T) {
+	t.Parallel()
+	boomIndex := "boomIndex"
+	namespaceIndex := "namespaceIndex"
+	namespaceVal := "namespace1"
+	n := 10 // Enough objects to trigger random map iteration order
```

Comment:
It will never flake - it is trying to detect if there is something that should not be there. So it can miss a bug (pass when it should not) but will never fail when everything is fine.